test/dashdb.dml.test.js

```diff
@@ -0,0 +1,65 @@
+// Copyright IBM Corp. 2019. All Rights Reserved.
+// Node module: loopback-connector-dashdb
+// This file is licensed under the Artistic License 2.0.
+// License text available at https://opensource.org/licenses/Artistic-2.0
+
+'use strict';
+
+/* eslint-env node, mocha */
+process.env.NODE_ENV = 'test';
+
+require('should');
+
+const Transaction = require('loopback-connector').Transaction;
+
+let db, FloatingPoint;
+
+describe('dml', function() {
+  before(function() {
+    require('./init.js');
+  });
+
+  describe('insert and patch', function() {
+    before(function(done) {
+      db = global.getDataSource();
+
+      FloatingPoint = db.define('FLOATING_POINT', {
+        id: {type: String, length: 20, index: true},
+        aFloat: {type: Number, dashdb: {dataType: 'double'}},
+      });
+      db.automigrate('FLOATING_POINT', done);
+    });
+
+    // Return an async function create a record in floating_point table
+    function createRecInFP(rec) {
+      return function(done) {
+        FloatingPoint.create(rec,
+          function(err, p) {
+            if (err) {
+              done(err);
+            } else {
+              done();
+            }
+          });
+      };
+    }
+
+    describe('patch', function() {
+      const rec = {aFloat: 42.234567};
+      before(createRecInFP(rec));
+
+      it('should fail with a specific SQL error', function(done) {
+        FloatingPoint.updateAll(rec, {aFloat: 42.234567890123456789},
+          function(err, info) {
+            if (err) {
+              err.state.should.be.eql('HY094');
```

Comment:
@rbetjes Is this test case expected to fail with error `HY094` or succeed? I think one test case can only have one expected behavior. 

From your code looks like you aim to test the error. And `info` should be `{count: 0}`, instead of throwing error due to `info[0]` is undefined.

Reply:
You're right. The else (line 56) in that if statement is never used. 
The update statement gives an error. Before the fix it results in an uncaught TypeError, aftewards it returns sql error state 'HY094'.
I'll fit it.

Reply:
@rbetjes
Almost there! one more thing(I should have posted it earlier):

If no error is thrown, the test should fail, could you use 
```ts
await FloatingPoint.updateAll(rec, {aFloat: 42.234567890123456789}).should.be.rejectedWith('HY094')
```
instead? 

Like https://github.com/strongloop/loopback-datasource-juggler/blob/2f867576b66fe535aa1f82bceaaa680ecd4eec6c/test/manipulation.test.js#L123-L124

If state 'HY094' is not included in the error msg, let's at least make sure the test fails when no error happens:

```ts
        FloatingPoint.updateAll(rec, {aFloat: 42.234567890123456789},
          function(err, info) {
            if (err) {
              err.state.should.be.eql('HY094');
              return done();
            }
            done(new Error('should fail with state HY094'));
          });
```